frontend/src/App.js

```diff
@@ -1,11 +1,49 @@
-import React from 'react';
-
-function App() {
-  return (
-    <div className="App">
-      <h1>Mafia Start</h1>
-    </div>
-  );
+import React from "react";
+import WelcomeMenu from "./WelcomeMenu";
+import WaitingRoom from "./WaitingRoom";
+const PHASE = {
+  START: 0,
+  WAITING_ROOM: 1,
+  IN_GAME: 2
+};
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      phase: PHASE.START
+    };
+
+    this.componentChoser = this.componentChoser.bind(this);
+    this.onNewGame = this.onNewGame.bind(this);
+  }
+
+  onNewGame() {
+    this.setState({ phase: PHASE.WAITING_ROOM });
+  }
+
+  componentChoser() {
+    let retComp = null;
+    switch (this.state.phase) {
```

Comment:
instead of switch case, send the onNewGame to the actual handler

Reply:
will be changed in the add-context branch later